### ISSUE TITLE
GHA: docker run rebuilds images

### ISSUE THREAD

**Issue by KrisThielemans** [SyneRBI/SIRF-SuperBuild#763]
Our GHA for the SIRF latest/devel looks as follows:
https://github.com/SyneRBI/SIRF-SuperBuild/blob/844ca0f18ed334d9af91a8b18ad4390ddad81c91/.github/workflows/docker_build.yml#L101-L119
Unfortunately, at every stage, the docker image gets rebuild.:
1. The expectation would be that once `core` is built, building `sirf` wouldn't need to run all the `apt` command, as `sirf` image essentially [starts with `user_python_ubunutu/sh`](https://github.com/SyneRBI/SIRF-SuperBuild/blob/844ca0f18ed334d9af91a8b18ad4390ddad81c91/docker/Dockerfile#L66
). All the `apt` commands get repeated though, see [here](https://github.com/SyneRBI/SIRF-SuperBuild/runs/7967521947?check_suite_focus=true#step:6:1). I guess the `docker load` statement was intended to prevent this, but it doesn't work.

2. we first build the image and then run it for the tests. But the rebuilding starts all over again, from `apt` and everything
https://github.com/SyneRBI/SIRF-SuperBuild/runs/7967521947?check_suite_focus=true#step:7:1). Weirdly, @casperdcl 's .travis.yml runs with exactly [the same line](https://github.com/SyneRBI/SIRF-SuperBuild/blob/844ca0f18ed334d9af91a8b18ad4390ddad81c91/.travis.yml#L160) and on Travis, all "core" lines in the Dockerfile [use the docker cache](https://app.travis-ci.com/github/SyneRBI/SIRF-SuperBuild/jobs/580577881#L907) as expected.

Obviously, this makes docker testing on GHA very slow/resource intensive.

@lauramurgatroyd any ideas?

**Comment by KrisThielemans** [comment]
Apologies, the first step (building of the `sirf` image) does work as expected, so I've edited the above post

**Comment by casperdcl** [comment]
Not sure I follow. It looks like it's using the cache to me: https://github.com/SyneRBI/SIRF-SuperBuild/runs/7904860811?check_suite_focus=true#step:6:111

**Comment by KrisThielemans** [comment]
> Not sure I follow. It looks like it's using the cache to me: https://github.com/SyneRBI/SIRF-SuperBuild/runs/7904860811?check_suite_focus=true#step:6:111

yeah. sorry. I corrected myself. It's only the running that rebuilds.

**Comment by lauramurgatroyd** [comment]
I don't know a lot about how the docker commands work, I just used the same ones as in the `travis.yml`

**Comment by KrisThielemans** [comment]
For whatever reason, this rebuild doesn't happen at the moment in #769, see e.g. https://github.com/SyneRBI/SIRF-SuperBuild/runs/8164071848?check_suite_focus=true#step:7:7. I can't say I understand this caching/rebuilding ...
